shared/markdown/generate-emoji-parser.js

```diff
@@ -35,9 +42,6 @@ function genEmojiData() {
       )
     }
     addEmojiLiteral(emoji.unified, emoji.short_name)
-    if (emoji.non_qualified) {
```

Comment:
Ah, looking at this again, I don't think this is correct. I'm guessing you're doing this because in
```
const emojiIndexByName = ${JSON.stringify(invert(emojiIndexByChar))}
```
the emoji name will point to the non-qualified char, which will cause problems on mobile.

But we also use `emojiIndexByChar` (and `emojiRegex`/`emojiExp`) to render `NativeEmoji` in `parser.pegjs`:
```
   while ((match = emojiExp.exec(emojiText)) !== null) {
     results.push(emojiText.substring(idx, match.index))
     results.push({type: 'native-emoji', children: [emojiIndexByChar[match[0]]]})
     idx = match.index + match[0].length
   }
```
so removing non-qualified versions means that we won't convert them to native-emoji objects, which seems wrong.

I think what we want to do is have separate `emojiIndexByCharMain` and `emojiIndexByCharNonQualified` lists, and have `emojiIndexByName` be the inverse of `emojiIndexByCharMain` (possibly barfing if there are duplicates), and then merge `emojiIndexByCharMain` and `emojiIndexByCharNonQualified` into `emojiIndexByChar`.

And I guess we should have tests for qualified and non-qualified native emojis turning into `native-emoji` objects with respective qualified versions (or do we always want to convert non-qualified to qualified? not sure.)

Reply:
About your last question, it looks like unicode leaves it up to [implementation](http://unicode.org/reports/tr51/#Emoji_Implementation_Notes) whether to render non-qualified versions as emoji. I think for display we should always use the qualified version, since that's the one iOS accepts for the more problematic ones. I'll change the maps so we'll also handle non-qualified on the input as well. ty for looking again!